napari/animation/animate.py

```diff
@@ -0,0 +1,303 @@
+                        states_dict.append(copy.deepcopy(empty))
+        for ind, x in enumerate(states_dict):
+            x["frame"] = ind
+        self.states_dict = states_dict
+
+    def create_steps(self):
+        """Interpolate states between key-frames"""
+
+        self.create_state_dict()
+        self.interpolated_states = util.interpolate(self.states_dict)
+
+    def key_interpolframe(self, viewer):
+        """Progress through interpolated frames"""
+
+        self.create_steps()
+
+        new_frame = (self.current_interpolframe + 1) % len(self.states_dict)
+        self.update_napari_state(new_frame)
+        self.current_interpolframe = new_frame
+
+    def collect_images(self):
+        """Collect images corresponding to all interpolated states
+
+        Returns
+        -------
+        image_stack : 3D numpy
+            stack of all snapshots
+        """
+
+        images = []
+        self.create_steps()
+        for i in range(len(self.states_dict)):
+
+            self.update_napari_state(i)
+            images.append(self.viewer.screenshot())
+
+        image_stack = np.stack(images, axis=0)
+        return image_stack
+
+    def update_napari_state(self, frame):
+        """Set the napari viewer to a given interpolated frame
+
+        Parameters
+        -------
+        frame : int
+            frame to visualize
+        """
+
+        # set view type 2D/3D and camera state
+        self.viewer.dims.ndisplay = self.interpolated_states["ndisplay"][frame]
+        self.viewer.window.qt_viewer.view.camera.set_state(
+            self.interpolated_states["camera"][frame]
+        )
+
+        # assign interpolated visibility state
+        for j in range(len(self.viewer.layers)):
+            self.viewer.layers[j].visible = self.interpolated_states["vis"][
+                frame
+            ][j]
+
+        # adjust slider positions
+        for i in range(self.interpolated_states["sliders"].shape[1]):
+            self.viewer.dims.set_point(
+                i, self.interpolated_states["sliders"][frame][i]
+            )
+
+        # update view
+        self.viewer.window.qt_viewer.view.camera.view_changed()
+
+    def make_movie(self, name="movie.mp4", fps=20, quality=5):
+        """Create a movie based on key-frames selected in napari
+
+        Parameters
+        -------
+        name : str
+            name to use for saving the movie (can also be a path)
+            should be either .mp4 or .gif
+        fps : int
+            frames per second
+        quality: float
+            number from 1 (lowest quality) to 9
+            only applies to mp4
+        """
+
+        # creat all states
+        self.create_steps()
+
+        # create imageio writer and add all frames
+        _, extension = os.path.splitext(name)
+        if extension == '.mp4':
+            writer = imageio.get_writer(name, fps=fps, quality=quality)
+        else:
+            writer = imageio.get_writer(name, fps=fps)
```

Comment:
if you pass `quality` to a non-mp4 `get_writer` does it error or just get ignored? If ignored can we always pass, as it will make our code a little cleaner

Reply:
I had tried this and it generates an error, which is why I have this condition. But I can again split this into two functions if you think that's cleaner.

Reply:
keeping as one function is better, but I'd like to mirror the api of `get_writer` as much as possible, and I don't really like splitting based on the extension, which seems brittle and special case like.

What happens if you try and make an mp4 without setting quality? Is `5` the default value inside imageIO too?

If so we could make `quality=None` be default and then splitting on `if quality is not None:` that means if there is a non mp4 writer that uses `quality` we now have access to that too, if someone tries to make an mp4 without setting quality things will still be ok, and if they want to set quality they can too.

How does this sound?

Reply:
As suggested the writer is now create differently depending on whether a ```quality``` option is passed and not on the extension type.